website/www/site/content/en/documentation/sdks/java/testing/tpcds.md

```diff
@@ -0,0 +1,183 @@
+
+## TPC-DS extension in Beam
+
+### Reasons
+
+Beam provides a simplified implementation of TPC-DS benchmark and there are several reasons to have it in Beam:
+
+- Compare the performance boost or degradation of Beam SQL for different runners or their versions
+- Run Beam SQL on different runtime environments
+- Detect missing Beam SQL features or incompatibilities
+- Find performance issues in Beam
+
+### Queries
+All TPC-DS queries in Beam are pre-generated and stored in the provided artifacts.
+
+For the moment, 28 out of 103 SQL queries (99 + 4) successfully pass by running with Beam SQL transform since not all
+SQL-99 operations are supported.
+
+Currently supported queries:
+ - 3, 7, 10, 22, 25, 26, 29, 35, 38, 40, 42, 43, 50, 52, 55, 69, 78, 79, 83, 84, 87, 93, 96, 97, 99
+
+### Tables
+All TPC-DS table schemas are stored in the provided artifacts.
+
+### Input data
+Input data is already pre-generated for two data formats (CSV and Parquet) and stored in Google Cloud Storage (gs://beam-tpcds)
+
+### Runtime
+TPC-DS extension for Beam can only be run in **Batch** mode and supports these runners for the moment:
+- Spark Runner
+- Flink Runner
+- Dataflow Runner
+
+## TPC-DS output
+
+TBD
+
+## Benchmark launch configuration
+
+The TPC-DS launcher accepts the `--runner` argument as usual for programs that
+use Beam PipelineOptions to manage their command line arguments. In addition
+to this, the necessary dependencies must be configured.
+
+When running via Gradle, the following two parameters control the execution:
+
+    -P tpcds.args
+        The command line to pass to the TPC-DS main program.
+
+    -P tpcds.runner
+	The Gradle project name of the runner, such as ":runners:spark:3" or
+	":runners:flink:1.13. The project names can be found in the root
+        `settings.gradle.kts`.
+
+Test data has to be generated before running a suite and stored to accessible file system. The query results will be written into output files.
+
+### Common configuration parameters
+
+Size of input dataset (1GB / 10GB / 100GB / 1000GB):
+
+    --dataSize=<1GB|10GB|100GB|1000GB>
+
+Path to input datasets directory:
+
+    --dataDirectory=<path to dir>
+
+Path to results directory:
+
+    --resultsDirectory=<path to dir>
+
+Format of input files:
+
+    --sourceType=<CSV|PARQUET>
+
+Run queries (comma separated list of query numbers or `all` for all queries):
+
+    --queries=<1,2,...N|all>
+
+Number of queries **N** that are running in parallel:
+
+    --tpcParallel=N
+
+## Running TPC-DS
+
+There are some examples how to run TPC-DS benchmark on different runners.
+
+Running suite on the SparkRunner (local) with Query3 against 1Gb dataset in Parquet format:
+
+    ./gradlew :sdks:java:testing:tpcds:run \
+        -Ptpcds.runner=":runners:spark:3" \
+        -Ptpcds.args="
+            --runner=SparkRunner
+            --dataSize=1GB
+            --sourceType=PARQUET
+            --dataDirectory=gs://beam-tpcds/datasets/parquet/partitioned
```

Comment:
Could/should we resolve this path given dataSize and sourceType? (we could still allow the user to override it)